### ISSUE TITLE
Generic/ScopeIndent: false positive with nested closures

### ISSUE THREAD

**Issue by jrfnl** [PHPCSStandards/PHP_CodeSniffer#1312]
## Describe the bug

The `Generic.WhiteSpace.ScopeIndent` sniff incorrectly flags code _within_ a closure, and directly following a _nested_ closure as having the wrong indent, while the indent is correct.

### Code sample
Anonymized minimal reproduction sample:
```php
$obj->callA($foo)
    ->callB(function (string $bar) use ($baz) {
        $joined = function (array $newParts) {
            return implode('/', $newParts);
        };

        foreach ($something as $foobar) {
            return $foobar;
        }
    })
;
```

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php --standard=generic--sniffs=generic.whitespace.scopeindent`
3. See error message displayed
```text
------------------------------------------------------------------------------------------------------------------------
FOUND 2 ERRORS AFFECTING 2 LINES
------------------------------------------------------------------------------------------------------------------------
  9 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 8 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
 11 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 8 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
------------------------------------------------------------------------------------------------------------------------
```
☝🏻 Line 9 is the `foreach` line, line 11, the line with the closing brace for the `foreach`.

## Expected behavior

No errors from this sniff.

## Versions (please complete the following information)

|                                          |                                                                                                                |
|-----------------------------|---------------------------------------------------------------------------- |
| Operating System            | not relevant                                                           |
| PHP version                      | not relevant                                                                                          |
| PHP_CodeSniffer version | `4.x`                                                                                  |
| Standard                          | `Generic`                                                          |
| Install type                       | git clone |

## Please confirm

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/HEAD/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `4.x` branch of PHP_CodeSniffer.

**Comment by rodrigoprimo** [comment]
A similar request was previously discussed in the original repository: squizlabs/PHP_CodeSniffer#3199. Linking for visibility.

**Comment by Amoifr** [comment]
Triage notes. Still reproducible on both `3.x` and `4.x` at HEAD, and on the 4.0.4 release.

**The fixer makes it worse.** `phpcbf` does not just report the wrong indent, it applies it, turning correct code into incorrect code:

```diff
-        foreach ($something as $foobar) {
-            return $foobar;
-        }
+    foreach ($something as $foobar) {
+        return $foobar;
+    }
```

**The trigger is narrower than the sample suggests.** Six variants of the reported snippet:

| Variant | Result |
| --- | --- |
| Chained call, no nested closure | clean |
| Nested closure, no chaining (`$obj->callB(function () {`) | clean |
| Chaining, plain variable in place of the nested closure | clean |
| Chaining, nested closure passed as an argument (`}, $baz);`) | clean |
| Chaining, nested arrow function | clean |
| **Chaining, nested closure assigned** (`};`) | **2 errors** |

The leading `->callA($foo)` is not needed either: `$obj` alone followed by a single chained call reproduces it. So the condition is a closure that opens on a **continuation line of a method chain**, containing a nested closure whose statement ends in `};`.

**Where it goes wrong.** With `--runtime-set scope_indent_debug 1`, the nested closure's `}` sets the indent correctly, and then the "Close closure" branch overrides it:

```
=> indent set to 8 by token 57 (T_CLOSE_CURLY_BRACKET)
Close closure on line 6
        * token is inside condition 13 (T_CLOSURE) on line 3 *
        * first token on line 3 is 10 (T_OBJECT_OPERATOR) *
        * amended previous is T_VARIABLE on line 2 *
        * amended first token is 2 (T_VARIABLE) on line 2 *
        => indent set to 4 by token 2 (T_VARIABLE)
```

In `ScopeIndentSniff`, the enclosing closure is picked as `$prev`, and `$first` becomes the first token on its line, the `->` at column 5. The `findStartOfStatement()` amendment then walks that back to `$obj` at column 1 on the previous line, and `$currentIndent` is computed as `column - 1 + $this->indent`, giving 4 instead of 8.

That amendment is what loses the information: the anchor should stay on the line where the closure actually opens, which carries the chain's continuation indent. Without it, `->` at column 5 gives `4 + 4 = 8`, the correct value.

It also explains the variants. Without chaining, the closure opens on the first line of its own statement, so the amendment is a no-op and the result is right by accident. A nested closure passed as an argument or an arrow function never reaches this branch at all.

Reproduced with PHP 8.4 on Linux. Happy to test a patch against these variants.
